### ISSUE TITLE
U3 update to Warschauer Str.

### ISSUE THREAD

**Issue by a2** [derhuerst/bvg-topological-map#7]
I would love to make a PR that updates the U3 line to Warschauer Str. but I don't know how this all works. Is there a guide for proposing updates to the JSON file?

**Comment by derhuerst** [comment]
Some context: I chose the JSON to ensure maximum reusability for `bvg-topological-map`. With the given markup, both SVG-native tools like browser libs can render SVG from it, but other tools can convert it into other formats like PostScript, because [SVG path strings](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d#Path_commands) are well-supported.

---

Unfortunately, there is no guide currently. I guess it comes down to how much time you want to invest.

**This repo is AFAIK the only machine-readable representation of the BVG transit map, so it's worth a lot to keep it up to date!** However, the existing JSON file is the result of a time-intensive manual process:

- download the transit map as a PDF
- import in Sketch.app, which converts it to a SVG-like tree
- clean it up: remove labels, logos, etc
- join line shapes that are split into pieces, put their path definitions into the right order
- tag all stations with their station IDs, using [`vbb-stations-cli`](https://npmjs.com/package/vbb-stations-cli)
- export as SVG
- transform into JSON
- optional: add special information like wifi

**Repeating this process would yield a JSON file known to be up to date, but updating the JSON *directly* to reflect the changes seems more realistic.** You would have to convert the JSON into SVG, edit it and reflect the changes back into the JSON file.

As documented in [`vbb-map-routing#3`](https://github.com/derhuerst/vbb-map-routing/issues/12), I had started to update this repo to the latest transit map a while ago, but have stopped since, because I don't have enough time right now.

**Comment by derhuerst** [comment]
If you can think of a more straightforward way to keep this repo up to date, let me know!

**Comment by derhuerst** [comment]
Do give you an impression about the time required: It took me about 6 hours to take the official transit map through the whole process.

**Comment by derhuerst** [comment]
S-Bahn Berlin has published an [interactive network map](https://sbahn.berlin/liniennetz/) which has rasterised layers for all stations & lines. Let's see if we can reconstruct vector data from that!

**Comment by derhuerst** [comment]
[BVG has a new transit map, with the connected U5.](https://www.bvg.de/de/index.php?section=downloads&cmd=58&download=399)

**Comment by creativecreatorormaybenot** [comment]
> S-Bahn Berlin has published an interactive network map which has rasterised layers for all stations & lines. Let's see if we can reconstruct vector data from that!

@derhuerst did you end up experimenting with that?

**Comment by derhuerst** [comment]
No, I haven't worked on this so far. You're very welcome to do so if you want.
